logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/PersistMessageUseCase.kt

```diff
@@ -12,22 +16,34 @@ interface PersistMessageUseCase {
     suspend operator fun invoke(message: Message.Standalone): Either<CoreFailure, Unit>
 }
 
+@OptIn(DelicateKaliumApi::class)
 internal class PersistMessageUseCaseImpl(
     private val messageRepository: MessageRepository,
-    private val selfUser: UserId
```

Comment:
is there a reason for not using the cached self user id ?

Reply:
I think we need more info about self user in some functions

Reply:
SelfUserId is not enough, we needed the availability state of the selfUser.